### PR TITLE
GEOPY-1701: Add validation with mesh_type on ObjectForm

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-1701 - Add validation with mesh_type on ObjectForm**

### CODE REVIEW

File: geoh5py/ui_json/validations/uijson.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+def parent_validation(name: str, data: dict[str, Any], params: dict[str, Any]):
+    """Data is a child of the parent object."""
+
+    form = params[name]
+    child = data[name]
+    parent = data[form["parent"]]
+
+    if isinstance(child, Data):
+        if (
```
Comment (domfournier) [inline geoh5py/ui_json/validations/uijson.py:86]: Combine logic with `and`

File: tests/ui_json/uijson_test.py [hunk 2]
```diff
@@ -122,3 +122,58 @@ class MyUIJson(BaseUIJson):
 
     assert "my_absent_uid_parameter" in str(err.value)
     assert "my_faulty_data_parameter" in str(err.value)
+
+
+def test_validation(tmp_path):
+    from geoh5py.objects import Curve
```
Comment (domfournier) [inline tests/ui_json/uijson_test.py:128]: Imports on top

File: tests/ui_json/uijson_test.py [hunk 3]
```diff
@@ -122,3 +122,58 @@ class MyUIJson(BaseUIJson):
...
+    assert "my_data_parameter data is not a child of my_other_object_parameter" in str(
+        err.value
+    )
+    assert (
+        "Object's mesh type must be one of [<class 'geoh5py.objects.curve.Curve'>]"
+        in str(err.value)
+    )
+    assert "Dependency my_other_object_parameter must be either optional or" in str(
+        err.value
+    )
```
Comment (domfournier) [inline tests/ui_json/uijson_test.py:179]: Would you mind splitting this test into three? Will be easier to maintain and append later on. You could have some utility function that generates a UIJson class from a generic one, and just call it in each sub-test?

File: geoh5py/ui_json/validations/uijson.py
Comment (domfournier) [inline geoh5py/ui_json/validations/uijson.py:1]: What else is going to be added to this module? Could we just take over the existing validation file and eventually deprecate the InputValidation class?

**Review by domfournier — COMMENTED** [review]
Very nice, I like!
See comments.

File: geoh5py/ui_json/validations/uijson.py (hunk 1 above)
Comment (benk-mira) [reply geoh5py/ui_json/validations/uijson.py:86]: done.

File: tests/ui_json/uijson_test.py (hunk 2 above)
Comment (benk-mira) [reply tests/ui_json/uijson_test.py]: whoops - done.

File: tests/ui_json/uijson_test.py (hunk 3 above)
Comment (benk-mira) [reply tests/ui_json/uijson_test.py:191]: done.

File: geoh5py/ui_json/validations/uijson.py
Comment (benk-mira) [reply geoh5py/ui_json/validations/uijson.py:1]: I think we should store the form validations and update the forms to use the functional form of the field/model validators:
image

**Review by domfournier — APPROVED** [review]

**Pull request merged**